client/apollo/react/src/Message/MessageCommon.tsx

```diff
@@ -1,69 +1,66 @@
-import validationIcon from "@material-symbols/svg-400/outlined/check_circle-fill.svg";
-import errorIcon from "@material-symbols/svg-400/outlined/error-fill.svg";
-import warningIcon from "@material-symbols/svg-400/outlined/warning-fill.svg";
-import neutralIcon from "@material-symbols/svg-400/outlined/info-fill.svg";
-import infoIcon from "@material-symbols/svg-400/outlined/emoji_objects-fill.svg";
-import {
-  type ComponentPropsWithoutRef,
+import type {
+  ComponentPropsWithoutRef,
   ComponentType,
-  type PropsWithChildren,
-  type ReactElement,
-  useMemo,
+  ReactElement,
+  ReactNode,
 } from "react";
-
 import type { ButtonProps } from "../Button/ButtonCommon";
+import type { Icon } from "../Icon/IconCommon";
 import { Link } from "../Link/LinkCommon";
-import { Svg } from "../Svg/Svg";
+import { iconByVariant, messageVariants } from "./constants";
+import { getAriaRole } from "./Message.helpers";
+import type { MessageVariants } from "./types";
 
 type Headings = "h2" | "h3" | "h4" | "h5" | "h6";
 
-export const messageVariants = {
-  validation: "validation",
-  error: "error",
-  warning: "warning",
-  information: "information",
-  neutral: "neutral",
-} as const;
-
-export type MessageVariants = keyof typeof messageVariants;
-
 export type MessageProps = {
+  /** Message variant (validation, error, warning, information, neutral) */
   variant: MessageVariants;
+  /** Title displayed in the message */
   title?: string;
+  /** Main content of the message */
+  children?: ReactNode;
+  /** Action (link or button) displayed on the right side of the message */
   action?: ReactElement<typeof Link | ComponentType<ButtonProps>>;
+  /** Icon size in pixels */
   iconSize?: number;
+  /** HTML heading level used for the title */
   heading?: Headings;
-} & ComponentPropsWithoutRef<"div">;
+} & ComponentPropsWithoutRef<"section">;
+
+type MessagePropsWithComponents = MessageProps & {
+  IconComponent: typeof Icon;
+};
 
-const getIconFromType = (variant: MessageVariants) =>
-  ({
-    [messageVariants.validation]: validationIcon,
-    [messageVariants.error]: errorIcon,
-    [messageVariants.neutral]: neutralIcon,
-    [messageVariants.warning]: warningIcon,
-    [messageVariants.information]: infoIcon,
-  })[variant] || infoIcon;
+const defaultClassName = "af-message";
 
-export const Message = ({
+export const MessageCommon = ({
   variant = messageVariants.information,
   className,
   title,
   children,
   action,
   iconSize = 24,
   heading: Heading = "h4",
-}: PropsWithChildren<MessageProps>) => {
-  const icon = useMemo(() => getIconFromType(variant), [variant]);
+  IconComponent,
+  ...sectionProps
+}: MessagePropsWithComponents) => {
+  const role = getAriaRole(variant);
```

Comment:
Tu pourrais mettre le role en paramètre, cela permettrai quand même au développeur de pouvoir gérer les exceptions.

```suggestion
 export const MessageCommon = ({
  variant = messageVariants.information,
  className,
  title,
  children,
  action,
  iconSize = 24,
  heading: Heading = "h4",
  IconComponent,
  role = getAriaRole(variant),
  ...sectionProps
}: MessagePropsWithComponents) => {
 
```

Reply:
on pourra voir ça plus tard